File: services/client/src/packet/bet_packet.go
package packet

import (
	"github.com/7574-sistemas-distribuidos/tp-nivelador/src/lottery"
)

type BetPacket struct {
	bet lottery.Bet
}

func NewBetPacket(bet lottery.Bet) Packet {
	return &BetPacket{
		bet: bet,
	}
}

func betFromBytes(payload []byte) (Packet, error) {
	parsedBet, err := decodeBet(payload)
	if err != nil {
		return nil, err
	}

	return NewBetPacket(parsedBet), nil
}

func (bPkt *BetPacket) ToBytes() ([]byte, error) {
	payload, err := encodeBet(bPkt.bet)
	if err != nil {
		return nil, err
	}

	return buildFrame(BET_OPCODE, payload), nil
}

func (bPkt *BetPacket) Bet() lottery.Bet {
	return bPkt.bet
}

File: services/server/src/packet/bet_packet.py
from .packet import Packet, build_frame
from lottery import Bet
from .bet_codec import encode_bet, decode_bet
from .opcode import Opcode

class BetPacket(Packet):
    def __init__(self, bet: Bet):
        self.bet = bet

    def to_bytes(self) -> bytes:
        return build_frame(Opcode.BET, encode_bet(self.bet))

def bet_pkt_from_bytes(data: bytes) -> Packet:
    return BetPacket(decode_bet(data))